### PR TITLE
feat(tiering): MGET support

### PR DESCRIPTION
**Pull request by dranikpg**
Support MGET with tiering

### CODE REVIEW

File: src/server/string_family.cc [hunk 1]
```diff
@@ -417,28 +416,41 @@ SinkReplyBuilder::MGetResponse OpMGet(bool fetch_mcflag, bool fetch_mcver, const
...
   }
 
+  // Allocate enough for all values
   response.storage_list = SinkReplyBuilder::AllocMGetStorage(total_size);
   char* next = response.storage_list->data;
 
   for (size_t i = 0; i < iters.size(); ++i) {
     auto it = iters[i];
-    if (it.is_done())
+    if (it.is_done())  // might have been evicted due to reading other keys
```
Comment (romange) [inline src/server/string_family.cc:436]: can you explain this comment?

Comment (dranikpg) [reply src/server/string_family.cc:436]: Ok, might be wrong. I remember we had a bug around here with invalidation, there are just no comments about it 🤷🏻‍♂️

File: src/server/string_family.cc [hunk 2]
```diff
@@ -438,7 +438,19 @@ SinkReplyBuilder::MGetResponse OpMGet(bool fetch_mcflag, bool fetch_mcver, const
 
     auto& resp = response.resp_arr[i].emplace();
 
-    size_t size = CopyValueToBuffer(it->second, next);
+    // Copy to buffer or trigger tiered read that will eventually write to buffer
+    if (it->second.IsExternal()) {
+      wait_bc->Add(1);
+      auto cb = [next, wait_bc](const string& v) mutable {
```
Comment (romange) [inline src/server/string_family.cc:444]: nit: move wait_bc

File: src/server/tiered_storage.cc [hunk 3]
```diff
@@ -171,6 +171,16 @@ util::fb2::Future<string> TieredStorage::Read(DbIndex dbid, string_view key,
   return future;
 }
 
+void TieredStorage::Read(DbIndex dbid, std::string_view key, const PrimeValue& value,
+                         std::function<void(const std::string&)> readf) {
+  DCHECK(value.IsExternal());
+  auto cb = [readf = std::move(readf)](string* value) {
+    readf(*value);
+    return false;
```
Comment (romange) [inline src/server/tiered_storage.cc:179]: add a comment what false means

**Review by romange — APPROVED** [review]
👏🏼

File: src/server/tiered_storage.cc (hunk 3 above)
Comment (dranikpg) [reply src/server/tiered_storage.cc:179]: I have a comment on the callback type what false means, so I don't have to repeat it everywhere 🙂

Comment (romange) [reply src/server/tiered_storage.cc:179]: you have a typo there: `Returns whether the value was modified`. you mean `returns true if ....`

Comment (dranikpg) [reply src/server/tiered_storage.cc:179]: That's the same meaning, no? 🌹

**Pull request merged**